### ISSUE TITLE
A roots method for multivariate polynomials over finite fields

### ISSUE THREAD

**Issue by alexjbest** [oscar-system/Oscar.jl#2311]
**Is your feature request related to a problem? Please describe.**
It would be very convenient if Oscar had a method to find the roots of a multivariate polynomial over a finite field.
```julia
julia> R, (a, b,c) = PolynomialRing(FiniteField(7)[1], ["a", "b","c"])
(Multivariate Polynomial Ring in a, b, c over Galois field with characteristic 7, gfp_mpoly[a, b, c])

julia> roots(a^2 + b^2 + c^2)
ERROR: MethodError: no method matching roots(::gfp_mpoly)
```

**Describe the solution you'd like**
Even something relatively naive would be a good start, that could be improved upon later.
That said I imagine with singular and other libraries available it should be possible to do something more advanced without too much effort on the oscar side.
As far as I know flint does not have such functionality built in.

**Describe alternatives you've considered**
The user can of course do some really basic root searching easily.

**Comment by thofma** [comment]
I agree that it would be great to have this! I had a similar request (privately) not long ago.

**Comment by fieker** [comment]
On Mon, Apr 24, 2023 at 08:57:20AM -0700, Tommy Hofmann wrote:
> I agree that it would be great to have this! I had a similar request (privately) not long ago.
> 
> 
Wasn't this (badly) implemented for 0-dim ideals not too long ago?
My reollection from the magma days is that the best is try & error...

**Comment by fingolfin** [comment]
Is this something msolve could handle, @ederc ?

**Comment by fingolfin** [comment]
@alexjbest is this just about such simple examples? What concrete examples do you have in mind? Just trying to gauge if something simple would be sufficient for you or if you need world record class performance

**Comment by alexjbest** [comment]
@fingolfin, my original examples I was playing with are not so simple as the example above, but not much more complicated, say 3 or 4 variable multivariate polynomials, over smallish finite fields. So nothing world beating, but just something quicker than a naive search.

**Comment by ederc** [comment]
> Is this something msolve could handle, @ederc ?

No, msolve handles only zero dimensional systems.
